data/src/main/java/com/microsoft/azure/kusto/data/auth/ConnectionStringBuilder.java

```diff
@@ -497,6 +537,22 @@ public static ConnectionStringBuilder createWithAzureCli(String clusterUrl) {
         return csb;
     }
 
+    public static ConnectionStringBuilder createWithCustomTokenCredential(@NotNull String clusterUrl, @Nullable  TokenCredential tokenCredential, @Nullable TokenRequestContext tokenRequestContext) {
+        if (StringUtils.isEmpty(clusterUrl)) {
+            throw new IllegalArgumentException("clusterUrl cannot be null or empty");
+        }
+
+        if (tokenCredential == null) {
+            throw new IllegalArgumentException("tokenCredential cannot be null");
+        }
+
+        ConnectionStringBuilder csb = new ConnectionStringBuilder();
+        csb.clusterUrl = clusterUrl;
+        csb.customTokenCredential = tokenCredential;
+        csb.customTokenRequestContext = tokenRequestContext;
```

Comment:
we dont do that in other sdks .. we should consider this for all or none...

Reply:
Other SDKs don't have the same api

Reply:
I'm personally on the side of, make the user do auth themselves. It's more flexible. 

In my Azure ID branch I basically made a way for them to do the Azure Identity token fetching themselves because, you never know what the user is gonna want to do, and I couldn't think of a more flexible solution than that.

Doing it for the users is nice of y'all, and you can assume a good bit based on the connection string but it leads to lots of code that you have to maintain. Lots of the other azure SDKs just make the users do it themselves.

Reply:
I'd recommend the same approach in your other SDKs too. Less complicated auth code to maintain and change over time if you lean into sub-letting that piece out to azure-core